herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -3965,13 +3966,29 @@ private Record fetchRecord(Bytes key, Long pageId, LocalScanPageCache localScanP
                 LOGGER.log(Level.FINE, "table " + table.name + ", activePages " + pageSet.getActivePages() + ", record " + key + " deleted during data access");
                 return null;
             }
+            if (trialPages == null) {
+                trialPages = new long[maxTrials];
+            }
+            trialPages[trialPages.length - maxTrials] = pageId;
             pageId = relocatedPageId;
             if (maxTrials-- == 0) {
```

Comment:
shouldn't this be --maxTrials?

Reply:
Yes I missread the maxTrials-- check. It should be --maxTrials. To be conservative we'll change maxTrials from 2 to 3